pkg/download/downloader.go

```diff
@@ -89,8 +89,9 @@ func extractZIP(targetDir string, read io.ReaderAt, size int64) error {
 }
 
 // extractTARGZ extracts a gzipped tar file into the target directory.
-func extractTARGZ(targetDir string, in io.Reader) error {
+func extractTARGZ(targetDir string, at io.ReaderAt, size int64) error {
 	glog.V(4).Infof("tar: extracting to %q", targetDir)
+	in := io.NewSectionReader(at, 0, size)
```

Comment:
IMO this change is unnecessary, you should just ignore `size` by naming it as `_`. We're never not read a tar file from a reader's only first N bytes for any reason.